core/controlplane/config/templates/cloud-config-worker

```diff
@@ -845,3 +860,414 @@ write_files:
+        fi
+        exit
+      }
+
+      set -e
+      trap onerr ERR
+      trap onexit exit
+
+      if [ ! -f ${DEV_CONTAINER} ]
+      then
+        echo Downloading CoreOS ${COREOS_TRACK} developer image ${COREOS_VERSION}
+        SITE=${COREOS_TRACK}.release.core-os.net/amd64-usr
+        curl -s -L https://${SITE}/${COREOS_VERSION}/coreos_developer_container.bin.bz2 \
+          -z ${DEV_CONTAINER}.bz2 \
+          -o ${DEV_CONTAINER}.bz2
+        echo Decompressing
+        bunzip2 -k ${DEV_CONTAINER}.bz2
+      fi
+
+      if [ ! -f ${DRIVER_ARCHIVE_PATH} ]
+      then
+        echo Downloading NVIDIA Linux drivers version ${DRIVER_VERSION}
+        mkdir -p nvidia_installers
+        SITE=us.download.nvidia.com/XFree86/Linux-x86_64
+        curl -s -L http://${SITE}/${DRIVER_VERSION}/${DRIVER_ARCHIVE}.run \
+          -z ${DRIVER_ARCHIVE_PATH} \
+          -o ${DRIVER_ARCHIVE_PATH}
+      fi
+
+      rm -Rf ${PWD}/tmp
+      mkdir -p ${PWD}/tmp ${WORK_DIR}
+      cp -ul ${DRIVER_ARCHIVE_PATH} ${WORK_DIR}
+
+      cd ${WORK_DIR}
+      chmod +x ${DRIVER_ARCHIVE}.run
+      sudo rm -Rf ./${DRIVER_ARCHIVE}
+      ./${DRIVER_ARCHIVE}.run -x -s
+      cd ${ORIGINAL_DIR}
+
+      systemd-nspawn -i ${DEV_CONTAINER} \
+        --bind=${PWD}/_container_build.sh:/build.sh \
+        --bind=${PWD}/${WORK_DIR}:/nvidia_installers \
+        /bin/bash -x /build.sh ${EMERGE_SOURCES} ${DRIVER_VERSION} || echo "nspawn fails as expected.  Because kernel modules can't install in the container"
+
+      sudo chown -R ${UID}:${GROUPS[0]} ${PWD}/${WORK_DIR}
+
+      bash -x _export.sh ${WORK_DIR}/*-${DRIVER_VERSION} \
+        ${DRIVER_VERSION} ${COREOS_VERSION}-${DRIVER_VERSION}
+
+  - path: /opt/nvidia-build/create-uvm-dev-node.sh
+    owner: root:root
+    permissions: 0755
+    content: |
+      #!/bin/sh
+      # This script is borrowed from https://github.com/Clarifai/coreos-nvidia/pull/4
+      # Get the major device number for nvidia-uvm and create the node
+      echo "Set up NVIDIA UVM"
+      major=`grep nvidia-uvm /proc/devices | awk '{print $1}'`
+      if [ -n "$major" ]; then
+          mknod -m 666 /dev/nvidia-uvm c $major 0
+      fi
+
+  - path: /opt/nvidia-build/nvidia-insmod.sh
+    owner: root:root
+    permissions: 0755
+    content: |
+      #!/bin/sh
+      # This script is borrowed from https://github.com/Clarifai/coreos-nvidia/pull/4
+      /usr/sbin/insmod /opt/nvidia/current/lib/modules/$(uname -r)/video/$1
+
+  - path: /opt/nvidia-build/nvidia-persistenced.service
+    owner: root:root
+    permissions: 0644
+    content: |
+      [Unit]
+      Description=NVIDIA Persistence Daemon
+      Wants=local-fs.target
+
+      [Service]
+      Type=forking
+      ExecStart=/opt/nvidia/current/bin/nvidia-persistenced --user nvidia-persistenced --no-persistence-mode --verbose
+      ExecStopPost=/bin/rm -rf /var/run/nvidia-persistenced
+
+  - path: /opt/nvidia-build/nvidia-start.service
```

Comment:
please see [the comment](https://github.com/kubernetes-incubator/kube-aws/pull/645/files#r116218676).  `nvidia-install.sh` does.